include/osquery/events.h

```diff
@@ -448,6 +448,28 @@ class EventSubscriberPlugin : public Plugin {
    */
   Status recordEvent(EventID& eid, EventTime time);
 
+  /**
+   * @brief Get the expiration timeout for this event type
+   *
+   * The default implementation retrieves this value from FLAGS_events_expiry.
+   * This method can be overridden to allow custom event expiration timeouts in
+   * subclasses of EventSubscriberPlugin.
+   *
+   * @return The events expiration timeout for this event type
+   */
+  virtual uint64_t getEventsExpiry();
```

Comment:
I think the answer to your second question is, yes. We had discussed that we didn't want the sylog table to have the same retention as other event based tables, and it seemed like a clean way to implement that by taking advantage of the existing logic and refactoring this so that we could add separate CLI flags. Do you have other ideas on how to do that in a DRY style?